crates/bevy_ecs/src/system/system_chaining.rs

```diff
@@ -142,3 +142,156 @@ where
         }
     }
 }
+
+/// A collection of common adapters for [chaining](super::ChainSystem) the result of a system.
+pub mod adapter {
```

Comment:
Before we go down this path: do we have real world use cases for these adapters?

For some system:

```rust
fn do_something(q: Query<&Transform>) -> Result<String, Error> {
  if SOME_CONDITION {
    Ok("hello there".to_string())
  } else {
    Err(Error::new())
  }
}
```

If the goal is to convert this to an `Option<String>`, by default shouldn't the solution be to return `Option<String>` in do_something?:

```rust
fn do_something(q: Query<&Transform>) -> Option<String> {
  if SOME_CONDITION {
    Some("hello there".to_string())
  } else {
    None
  }
}
```

Obviously there are differences here, as we've eliminated functionality from `do_something` that might be useful elsewhere. _In theory_ this enables more flexible scenarios. But I havent personally encountered a case that wouldn't be better off by embedding the logic in the system.

In practice, has anyone hit a case that benefits from the `ok` adapter in a real game? What does it look like?

The adapters that I have seen a use case for are error handler adapters that print the error in `Result<(), Error>`, if it exists.  This enables you to use `?` syntax in systems. Same goes for the `ignore` adapter (which you have included and which I can get behind).

Reply:
I can definitely remove the trivial adapters -- we can re-add them later if people come up with a use-case.

IMO, the must-have ones are `ignore` and `unwrap`, plus some `tracing` ones (if I knew how that crate worked). I could see `ok_or{_else}` being useful occasionally, but they're not super important. I'm not attached to the rest of them.

Thoughts?

Reply:
Ignore, unwrap and the tracing variants are the key ones for me.